ehrql/query_language.py

```diff
@@ -50,6 +50,23 @@ class DummyDataConfig:
     timeout: int = 60
     additional_population_constraint: "qm.Series[bool] | None" = None
 
+    def validate_additional_population_constraint(self):
+        if self.additional_population_constraint is not None:
+            validate_patient_series_type(
+                self.additional_population_constraint,
+                types=[bool],
+                context="additional population constraint",
+            )
+            self.additional_population_constraint = (
+                self.additional_population_constraint._qm_node
+            )
```

Comment:
I don't think this is quite right because now the "validate" method is changing the value of `self.additional_population_constraint`. So, for example, it will fail if you call it twice in a row.

I think you want to make it a `set_additional_population_constraint()` method which takes the value, validates it, and then updates the attribute if it's valid.

Reply:
Huh that's how I did it first. Will change it back, hadn't considered calling it twice